src/input/type_base.hh

```diff
@@ -297,6 +322,12 @@ public:
      */
     virtual bool valid_default(const string &str) const;
 
+    // Implements @p TypeBase::make_instance.
+    virtual MakeInstanceReturnType make_instance(std::vector<ParameterPair> vec = std::vector<ParameterPair>()) const;
```

Comment:
use override insead of virtual here and for other implementations of virtual methods.